### PR TITLE
bpo-46528: Simplify `BUILD_TUPLE`/`UNPACK_SEQUENCE` folding

### PR DESCRIPTION
**Pull request by brandtbucher**
Also, fix `test_pack_unpack`.

https://bugs.python.org/issue46528

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -8650,29 +8650,22 @@ optimize_basic_block(struct compiler *c, basicblock *bb, PyObject *consts)
...
+                   Replace BUILD_TUPLE(2) UNPACK_SEQUENCE(2) with SWAP(2).
+                   Replace BUILD_TUPLE(3) UNPACK_SEQUENCE(3) with SWAP(3). */
             case BUILD_TUPLE:
                 if (nextop == UNPACK_SEQUENCE && oparg == bb->b_instr[i+1].i_oparg) {
                     switch(oparg) {
                         case 1:
                             inst->i_opcode = NOP;
                             bb->b_instr[i+1].i_opcode = NOP;
-                            break;
+                            continue;
```
Comment (markshannon) [inline Python/compile.c:8662]: Why change this? Aren't they equivalent.
Having a `continue` here distracts from the flow of the code, as it is not the normal way to end a `case` and increases cognitive load (at least it did for me).

**Review by markshannon — CHANGES_REQUESTED** [review]
One nit about `continue` vs `break`. Otherwise LGTM.

File: Python/compile.c (hunk 1 above)
Comment (brandtbucher) [reply Python/compile.c:8662]: They’re not quite equivalent: note that I also removed the `break` on old line 8675.

Previously, values other than 1, 2, or 3 wouldn’t get a chance to hit `fold_tuple_on_constants`. Now they do.

Comment (markshannon) [reply Python/compile.c:8662]: :+1: LGTM then.

**Review by markshannon — APPROVED** [review]

**Pull request merged**